src/hooks/screen/useChatScreenHook.ts

```diff
@@ -0,0 +1,153 @@
+} from '../../utils/constants';
+import {getAnonymousUserId, getUserId} from '../../utils/users';
+import {randomString} from '../../utils/string/StringUtils';
+
+function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
+  const {localDb, chat, refresh} = useLocalDatabaseHook();
+  const {selectedChannel, goBackFromChatScreen, goToChatInfoScreen} = useChatUtilsHook();
+
+  const [chats, setChats] = React.useState<ChatSchema[]>([]);
+  const {anon_user_info_emoji_name} = selectedChannel?.rawJson?.channel || {};
+  const initChatData = async () => {
+    if (!localDb && !selectedChannel) return;
+    try {
+      const myUserId = await getUserId();
+      const myAnonymousId = await getAnonymousUserId();
+      const data = (await ChatSchema.getAll(
+        localDb,
+        selectedChannel?.id,
+        myUserId,
+        myAnonymousId
+      )) as ChatSchema[];
+      setChats(data);
+    } catch (e) {
+      console.log(e, 'error get all chat');
+    }
+  };
+
+  const sendChat = async (
+    message: string = randomString(20),
+    iteration = 0,
+    sendingChatSchema: ChatSchema | null = null
+  ) => {
+    if (iteration > 5) {
+      SimpleToast.show("Can't send message, please check your connection");
+      return;
+    }
+
+    let currentChatSchema = sendingChatSchema;
+    let userId = await getUserId();
+    const myAnonymousId = await getAnonymousUserId();
+
+    if (type === 'ANONYMOUS') {
+      userId = myAnonymousId;
+    }
+    try {
+      const randomId = uuid();
+
+      if (currentChatSchema === null) {
+        currentChatSchema = await ChatSchema.generateSendingChat(
+          randomId,
+          userId,
+          selectedChannel?.id,
+          message,
+          localDb
+        );
+        currentChatSchema.save(localDb);
+        refresh('chat');
+        refresh('channelList');
+      }
+      let channelType = CHANNEL_TYPE_PERSONAL;
+      if (
+        selectedChannel?.channelType?.toLowerCase() === 'group' ||
+        selectedChannel?.rawJson?.channel?.type === 'group'
+      ) {
+        channelType = CHANNEL_TYPE_GROUP;
+      }
+
+      if (selectedChannel?.rawJson?.channel?.channel_type === 4) {
+        channelType = CHANNEL_TYPE_ANONYMOUS;
+      }
+
+      let response;
+      if (type === 'ANONYMOUS') {
+        response = await AnonymousMessageRepo.sendAnonymousMessage(selectedChannel?.id, message);
+      } else {
+        response = await SignedMessageRepo.sendSignedMessage(
+          selectedChannel?.id,
+          message,
+          channelType
+        );
+      }
+      await currentChatSchema.updateChatSentStatus(localDb, response);
+      refresh('chat');
+      refresh('channelList');
+    } catch (e) {
+      if (e?.response?.data?.status === 'Channel is blocked') return;
+
+      setTimeout(() => {
+        sendChat(message, iteration + 1, currentChatSchema).catch((sendChatError) => {
+          console.log(sendChatError);
+        });
+      }, 1000);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling here could be improved. If the channel is blocked, it would be helpful to inform the user about this instead of failing silently.
```diff
- if (e?.response?.data?.status === 'Channel is blocked') return;
+ if (e?.response?.data?.status === 'Channel is blocked') {
+   SimpleToast.show("The channel is blocked. You cannot send messages.");
+   return;
+ }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->